dashboard/app/models/pd/application/teacher1819_application.rb

```diff
@@ -426,6 +445,43 @@ def check_idempotency
       Pd::Application::Teacher1819Application.find_by(user: user)
     end
 
+    def find_default_workshop
+      return unless regional_partner
+
+      workshop_course =
+        if course == 'csd'
+          Pd::Workshop::COURSE_CSD
+        elsif course == 'csp'
+          Pd::Workshop::COURSE_CSP
+        end
+
+      # If this application is associated with a G3 partner who in turn is
+      # associated with a specific teachercon, return the workshop for that
+      # teachercon
+      if regional_partner.group == 3
+        teachercon = get_matching_teachercon(regional_partner)
+        if teachercon
+          return find_teachercon_workshop(workshop_course, teachercon, 2018)
+        end
+      end
+
+      # Default to just assigning whichever of the partner's eligible workshops
+      # is scheduled to start first. We expect to hit this case for G1 and G2
+      # partners, and for any G3 partners without an associated teachercon
+      regional_partner.
+        pd_workshops_organized.
+        where(
+          course: workshop_course,
+          subject: [
+            Pd::Workshop::SUBJECT_TEACHER_CON,
+            Pd::Workshop::SUBJECT_FIT,
```

Comment:
I don't think we want FIT workshops included here for teacher application, do we? My understanding is those are only for facilitators, i.e. facilitator application.